actors.md

```diff
@@ -708,3 +708,233 @@ func UpdatePeerID(pid PeerID) {
+| -------- | ---------- | ------------------------------------------------------------ |
+| `signers`   | `[]Address`      | The addresses that will be the signatories of this wallet. |
+| `required` | `uint` | The number of signatories required to perform a transaction.       |
+
+```go
+func Multisig(signers []Address, required uint) {
+    self.Signers = signers
+    self.Required = required
+}
+```
+
+
+
+### Propose
+
+> Propose is used to propose a new transaction to be sent by this multisig. The proposer must be a signer, and the proposal also serves as implicit approval from the proposer. If only a single signature is required, then the transaction is executed immediately.
+
+| Name     | Type       | Description  |
+| --- | --- | --- |
+| `to` |  `Address` | The address of the target of the proposed transaction. |
+|  `value` | `TokenAmount` | The amount of funds to send with the proposed transaction |
+|  `method` | `string` | The method that will be invoked on the proposed transactions target. |
+| `params` | `[]byte` | The parameters that will be passed to the method invocation on the proposed transactions target. |
+
+```go
+func Propose(to Address, value TokenAmount, method string, params []byte) uint64 {
+    if !isSigner(msg.From) {
+        Fatal("not authorized")
+    }
+
+    txid := self.NextTxID
+    self.NextTxID++
+
+    tx := Transaction{
+        TxID: txid,
+        To: to,
+        Value: value,
+        Method: method,
+        Params: params,
+        Approved: []Address{msg.From},
+    }
+
+    self.Transactions.Append(tx)
+
+  if self.Required == 1 {
+    Send(tx.To, tx.Value, tx.Method, tx.Params)
+    tx.Complete = true
+  }
+
+    return txid
+}
+```
+
+### Approve
+
+> Approve is called by a signer to approve a given transaction. If their approval pushes the approvals for this transaction over the threshold, the transaction is executed.
+
+| Name     | Type       | Description  |
+| --- | --- | --- |
+| `txid` |  `uint64` | The ID of the transaction to approve. |
+
+
+```go
+func Approve(txid uint64) {
+    if !isSigner(msg.From) {
+        Fatal("not authorized")
+    }
+
+    tx := getTransaction(txid)
+    if tx.Complete {
+        Fatal("transaction already completed")
+    }
+    if tx.Canceled {
+        Fatal("transaction canceled")
+    }
+
+    for _, signer := range tx.Approved {
+        if signer == msg.From {
+            Fatal("already signed this message")
+        }
+    }
+
+    tx.Approved.Append(msg.From)
+
+    if len(tx.Approved) >= self.Required {
+        Send(tx.To, tx.Value, tx.Method, tx.Params)
+        tx.Complete = true
+    }
+}
+```
+
+### Cancel
+
+```go
+func Cancel(txid uint64) {
+    if !isSigner(msg.From) {
+        Fatal("not authorized")
+    }
+
+    tx := getTransaction(txid)
+    if tx.Complete {
+        Fatal("cannot cancel completed transaction")
+    }
+    if tx.Canceled {
+        Fatal("transaction already canceled")
+    }
+
+    tx.Canceled = true
+}
+```
```

Comment:
Can you explain how this can happen?
This sounds like the "veto" power, if n of m required people sign, why would one of the remaining party have power to stop the transaction? this sounds like a setting where is not always desirable

Reply:
IF n have signed, transaction goes through, however someone can cancel ahead of then. This does mean that an evil signer can DoS all m-1 other parties, though the proposer can just remove them from the signer set and try again.